flink-runtime/src/main/java/org/apache/flink/runtime/externalresource/ExternalResourceUtils.java

```diff
@@ -116,10 +117,25 @@ private ExternalResourceUtils() {
         return externalResourceConfigs;
     }
 
+    /**
+     * Instantiate {@link StaticExternalResourceInfoProvider} for all of enabled external resources.
+     */
+    public static ExternalResourceInfoProvider createStaticExternalResourceInfoProviderFromConfig(
+            Configuration configuration, PluginManager pluginManager) {
+
+        final Map<String, Long> externalResourceAmountMap =
+                getExternalResourceAmountMap(configuration);
+        LOG.info("Enabled external resources: {}", externalResourceAmountMap.keySet());
```

Comment:
What about `getExternalResourceConfigurationKeys`? This method also logs the same statement. Admittedly this is the statement logged on the `JobManager` process and this one here on the `TM`. Maybe we can change what is logged to make the distinction a bit clearer.

Reply:
I was only concerned with a single process logging the same message for this PR.

Overall I think this whole class needs a phrasing/sanity pass, but it didn't seem worth it at this time because it is a niche feature at this time.
For example, an external resource can be logged as being enabled, even if the driver could not be found later on. This is obviously misleading, and one could argue that we maybe should fail if the resource could not be loaded.
Similar cases are the amount sanity checks: "The amount of the {} should be positive while finding {}. Will ignore that resource."

Reply:
Shall we create a follow up ticket for this?

Reply:
yes, I'll open one.

Reply:
https://issues.apache.org/jira/browse/FLINK-22240